src/components/Calculator.js

```diff
@@ -1,44 +1,72 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.displayValues = this.displayValues.bind(this);
+  }
+
+  handleClick(btnName) {
+    const newState = calculate(this.state, btnName);
+    this.setState(() => ({
+      total: newState.total,
+      next: newState.next,
+      operation: newState.operation,
+    }));
+  }
+
+  displayValues() {
+    const { total, next, operation } = this.state;
+    if (total && !next && !operation) return `${total}`;
+    if (!total && next && !operation) return `${next}`;
+    if (total && !next && operation) return `${total} ${operation}`;
+    if (total && next && operation) return `${next}`;
+    return '0';
   }
 
   render() {
+    /* const { total, next, operation } = this.state; */
```

Comment:
- [ ] Please remove commented code, it makes your codebase messy. You can use git history if you need to look back into a previous version of your application.